### PR TITLE
[Windows] Now FileManager.isExecutableFile(atPath:) checks if the file exists and is readable

### PR DESCRIPTION
**Pull request by pvieito**

### CODE REVIEW

**Comment by pvieito** [comment]
@compnerd

File: Foundation/FileManager+Win32.swift [hunk 1]
```diff
@@ -497,8 +502,7 @@ extension FileManager {
     }
 
     internal func _isExecutableFile(atPath path: String) -> Bool {
-        // FIXME(compnerd) is there some test that we can perform here?
-        return true
+        return _nonDirectoryFileExists(atPath: path) && _isReadableFile(atPath: path)
```
Comment (compnerd) [inline Foundation/FileManager+Win32.swift]: Is there a reason to create the `_nonDirectoryFileExists`?  I think that we can just inline it (unless you have another use that you have planned?).

Comment (pvieito) [reply Foundation/FileManager+Win32.swift]: No, mainly for potential future usage and better readability. If you prefer it, I could inline it.

Comment (compnerd) [reply Foundation/FileManager+Win32.swift]: Yeah, I think that the inline is going to be better for now.  We can extract it when needed.

Comment (pvieito) [reply Foundation/FileManager+Win32.swift]: Done!

**Comment by compnerd** [comment]
Could you please squash the changes @pvieito?

**Comment by pvieito** [comment]
@compnerd Squashed at #2133

**Pull request closed without merging**